testsuite/tests/letrec-check/basic.ml

```diff
@@ -115,7 +115,10 @@ val y : 'a list -> unit = <fun>
 (* this is accepted as all fields are overridden *)
 let rec x = { x with contents = 3 }  [@ocaml.warning "-23"];;
 [%%expect{|
-val x : int ref = {contents = 3}
+Line 1, characters 12-35:
+1 | let rec x = { x with contents = 3 }  [@ocaml.warning "-23"];;
+                ^^^^^^^^^^^^^^^^^^^^^^^
+Error: This kind of expression is not allowed as right-hand side of `let rec'
```

Comment:
The comment `(* this is accepted as all fields are overridden *)` above that test doesn't hold anymore.